File: PUGD-frontend/graphql/queries/acquisition/provider.js
import gql from "graphql-tag";

const GetAllProviders = gql`
  query {
    getallproviders {
      _id
      name
      adress
      phone
      email
    }
  }
`;

const GetOneProvider = gql`
  query($id: String!) {
    getoneprovider(id: $id) {
      _id
      account
      adress
      email
      establishement
      name
      phone
      website
    }
  }
`;

module.exports = {
  GetAllProviders,
  GetOneProvider,
};

File: PUGD-api/graphql/types/AcquisitionTypes/Provider.go
package AcquisitionTypes

import "github.com/graphql-go/graphql"

var ProviderType = graphql.NewObject(graphql.ObjectConfig{
	Name: "ProviderType",
	Fields: graphql.Fields{
		"_id": &graphql.Field{
			Type: graphql.ID,
		},
		"establishement": &graphql.Field{
			Type: graphql.String,
		},
		"name": &graphql.Field{
			Type: graphql.String,
		},
		"account": &graphql.Field{
			Type: graphql.String,
		},
		"adress": &graphql.Field{
			Type: graphql.String,
		},
		"phone": &graphql.Field{
			Type: graphql.String,
		},
		"email": &graphql.Field{
			Type: graphql.String,
		},
		"website": &graphql.Field{
			Type: graphql.String,
		},
	},
})
